include/wlr/types/wlr_output.h

```diff
@@ -270,4 +307,69 @@ enum wl_output_transform wlr_output_transform_invert(
 enum wl_output_transform wlr_output_transform_compose(
 	enum wl_output_transform tr_a, enum wl_output_transform tr_b);
 
+/*
+ * Presentation functions
+ */
+
+/*
+ * Sets the rectangle of the input image which will be output to the display,
+ * and the size of the destination rectangle.
+ *
+ * This is designed to match the wp_viewporter protocol interface.
+ *
+ * src_x, src_y, src_w and src_h control cropping.
+ * dest_w and dest_h control scaling.
+ *
+ * This is not applied until the next call to wlr_output_present.
+ *
+ * Not every backend supports this.
+ * TODO: Add a way to query if it's supported ahead of time.
```

Comment:
This function should probably return a `bool`

Reply:
Yeah, I was originally thinking that, but if this ends up being not supported, this affects how you're going to render things, and you'd probably call this function _after_ you've done that.
I think the way to check needs to be done earlier and without changing the output state.

Reply:
I see. Then just add a function that checks that `impl->set_viewport != NULL` and assert this in `wlr_output_set_viewport`?